docs/spec/BeakerSpec.tex

```diff
@@ -0,0 +1,1055 @@
+  additional keys) of $B$ is less than or equal to the highest writable address
+  of $A$.
+\end{itemize}
+
+\subsubsection{Error Codes}
+There are no specific error conditions for SSTORE, therefore only general system
+call errors are applicable.
+
+\subsection{Log System Call}
+
+\subsubsection{System Call Format}
+Type Value: 8
+
+Data: min(96 bytes) max(224 bytes)
+
+\begin{table}[H]
+  \caption{System call structure.}
+  \centering{}%
+  \begin{tabularx}{\textwidth}{l|X}
+    \hline
+    Byte Offset & Description \\
+    \hline
+    \hline
+    \texttt{0x00} & Capability index; 32 bytes \\
+    \texttt{0x20} & Number of topics (nTopics) \\
+    ... & Potentially topic value \#1; 32 bytes \\
+    ... & Potentially topic value \#2; 32 bytes \\
+    ... & Potentially topic value \#3; 32 bytes \\
+    ... & Potentially topic value \#4; 32 bytes \\
+    \texttt{0x40}+(nTopics $\times$ 0x20) & Log value 32 bytes \\
+    \hline
+    End &  \\
+    \hline
+  \end{tabularx}
+\end{table}
+
+\subsubsection{Capability Format}
+The Write capability includes between 0 and 4 values. Each value forces the use
+of a particular value. For example: if the capability has 1 value, then that
+means that the first topic in the log call must equal that. If it has 2 values
+then the first log topic must be the first of those values and the second topic
+must be the second of those values and so on. If there are no topics listed then
+there are no restrictions on what can be logged.
+
+This capability is 5 values. If the number of enforced topics is less than 4,
+then the unused values are undefined.
+
+\begin{table}[H]
+  \caption{Log capability format.}
+  \centering{}%
+  \begin{tabularx}{\textwidth}{c|c|X}
+    \hline
+    Key Offset & Value & Description\\
+    \hline
+    \hline
+    \texttt{0x00} & [0x00, 0x4] & Number of enforced topics; 32 bytes \\
+    \texttt{0x01} & $t_1$ & Potentially an enforced value for the first topic
+      \\
+    \texttt{0x02} & $t_2$ & Potentially an enforced value for the second topic
+      \\
+    \texttt{0x03} & $t_3$ & Potentially an enforced value for the third topic
+      \\
+    \texttt{0x04} & $t_4$ & Potentially an enforced value for the fourth topic
+      \\
+    \hline
+    End &  \\
+    \hline
+  \end{tabularx}
+\end{table}
+
+\subsubsection{Capability Subsets}
+As defined in Section~\ref{cap-subsets}, each capability has subsets and
+supersets.
+
+Given a Log capability $A$ and a proposed subset $B$, $B \subseteq A$ if and
+only if for each log topic of $A$ the topic is either undefined or equal to that
+of $B$.
+
+If $C_{t_i}$, where $i$ is $1$, $2$, $3$, or $4$, is the log topic $i$ of
+capability $C$, and $C_n$ is the number of defined topics:
+
+$$ \forall i \in \left\{1,2,\dots ,A_n\right\}. B_{t_i} = A_{t_i} $$
+
+\subsubsection{Error Codes}
+There are no specific error conditions for SSTORE, therefore only general system
+call errors are applicable.
+
+\subsection{Value Send System Call}
```

Comment:
Reading section `4.10 Value Send System Call`, it's not clear what the format of the tx#send capability is. Just to confirm our previous discussion, `tx#send` capability should only contain the target address. (We decided not to include eth/gas limits since they can be handled by a system procedure).

Reply:
So it either includes all addresses (0x00) or a single address (0x00).

Reply:
Just like `proc#send` this is missing the memory slice (64 bytes) payload for the target contract address.

Reply:
Ok, so the capability should be updated to include the possibility of any address, or single address (currently it is only any address). You're right, this is consistent with our discussion.

Given you're third comment, we should review external calls and make sure the plan is cogent. Upon revision, maybe we should keep the value send capability but remove the system call and replace it with a CALL system call.

Reply:
Or rather EXTERNAL CALL as it should be called. That might be worth a separate issue.

Reply:
@JakeOShannessy: So do you mean replacing the name from SEND to CALL? 

If that's what you mean, now that you mention this, `tx#send` should probably be renamed to `account#call` or `account#send` since this capability takes "control" over using the instance account rights and ether. 

For example, if the instance was assigned tokens from a ERC20 contract, an `account#call` cap to that token address would designate access to those assigned tokens.

Reply:
I'm obviously bike-shedding at this point 😅

Reply:
No I don't think you are! It's something we should really think about. In my mind I was just thinking about the sending of Ether, but that always just comes as a side effect of an EVM CALL opcode (or CALLCODE, but we should never do a CALLCODE in my opinion).

My argument is that the "sending" part of this operation should be relegated to not being the primary point. For a while I was just thinking of it is a pure sending operation, but of course that cannot happen in EVM, it always implies a CALL.

Even as I write this I'm less sure. Let's do this: I'll added the payload field, as with a normal procedure call (at least leaving the spec in a workable state), and we'll take all the other discussion over to #54.